File: verify_final.py
import asyncio
from playwright.async_api import async_playwright

async def verify_final():
    async with async_playwright() as p:
        browser = await p.chromium.launch()
        context = await browser.new_context()
        page = await context.new_page()

        # 1. Test Global Login UI on Index
        try:
            print("Checking Guest Index...")
            await page.goto("http://localhost:8000/index.html")
            await page.wait_for_timeout(1000)
            # Check news section (should be visible if backend returned data)
            news_section = await page.query_selector("#news-section")
            print(f"News section visible: {news_section is not None}")

            print("Mocking Student Login (10th Grade)...")
            await page.evaluate("localStorage.setItem('currentUser', JSON.stringify({userId: 'USR-10-A', nombre: 'Juan 10A', rol: 'Estudiante', grado: 'Décimo', seccion: 'A'}))")
            await page.reload()
            await page.wait_for_timeout(1000)

            # Check segmentation: Courses should show 10th grade
            courses_btn = await page.query_selector("button:has-text('Cursos')")
            if courses_btn:
                await courses_btn.hover()
                await page.wait_for_timeout(500)
                grades_menu = await page.inner_text("#desktop-grades-menu")
                print(f"Grades in menu: {grades_menu}")

        except Exception as e:
            print(f"Error checking Index/Segmentation: {e}")

        # 2. Test Pseudocode Arrays
        try:
            print("Checking Pseudocode Arrays...")
            await page.goto("http://localhost:8000/pseudocode.html")
            await page.wait_for_timeout(1000)

            # Set Strict Profile
            await page.click("div.menu-item:has-text('Herramientas')")
            await page.click("text=Modo Estricto")

            # Input code with Dimension and Assignment
            code = """Algoritmo TestArreglos
    Definir miArreglo como Entero;
    Dimension miArreglo[5];
    miArreglo[1] <- 10;
    Escribir miArreglo[1];
FinAlgoritmo"""

            await page.evaluate(f"document.getElementById('codeEditor').innerText = `{code}`")
            await page.evaluate("updateEditor()")

            # Run
            await page.click("button[title='Ejecutar']")
            await page.wait_for_timeout(2000)

            console_output = await page.inner_text("#consoleOutput")
            print(f"Console Output:\n{console_output}")

            await page.screenshot(path="verify_pseudocode_run.png")
        except Exception as e:
            print(f"Error checking Pseudocode Arrays: {e}")

        await browser.close()

if __name__ == "__main__":
    asyncio.run(verify_final())

File: verify_final.js
const { chromium } = require('playwright');
const path = require('path');

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage();

  // Mock data
  await page.addInitScript(() => {
    const mockUser = { userId: '12345', nombre: 'Estudiante Prueba', cargo: 'Estudiante', grado: 'Décimo', seccion: 'A' };
    localStorage.setItem('currentUser', JSON.stringify(mockUser));

    // Mock IndexedDB
    window.indexedDB.open = function() {
       return {
         onupgradeneeded: null,
         onsuccess: function() {
           this.result = {
             transaction: () => ({
               objectStore: () => ({
                 get: () => ({ onsuccess: function() { this.result = { activities: [
                   { id: '1', titulo: 'Tarea 1', asignatura: 'Programación', fecha_limite: '2023-01-01', entregada: false, descripcion: 'Desc 1', tipo: 'Tarea' },
                   { id: '2', titulo: 'Tarea 2', asignatura: 'Programación', fecha_limite: '2026-06-21', entregada: true, descripcion: 'Desc 2', tipo: 'Tarea' }
                 ] }; this.onsuccess(); } })
               }),
               oncomplete: null
             }),
             close: () => {}
           };
           this.onsuccess();
         }
       };
    };
  });

  await page.goto('file://' + path.resolve('student-dashboard.html'));
  await page.waitForTimeout(1000);

  // 1. Check Tabs
  console.log('Checking tabs...');
  await page.click('button:has-text("Progreso Académico")');
  await page.waitForTimeout(500);
  await page.screenshot({ path: 'verification/screenshots/3_progress_tab.png' });

  // 2. Go back to Activities and Expand
  await page.click('button:has-text("Actividades")');
  await page.waitForTimeout(500);

  const subjectCard = page.locator('.subject-card').first();
  await subjectCard.click();
  await page.waitForTimeout(1000); // Wait for expansion animation

  // 3. Try to click Entregar Tarea on an overdue task
  // We need to handle the dialog
  page.on('dialog', async dialog => {
    console.log('Dialog shown:', dialog.message());
    await page.screenshot({ path: 'verification/screenshots/4_late_dialog.png' });
    await dialog.accept();
  });

  console.log('Clicking Entregar Tarea...');
  const entregaBtn = page.locator('button:has-text("Entregar Tarea")').first();
  if (await entregaBtn.isVisible()) {
      await entregaBtn.click();
      await page.waitForTimeout(1000);
      await page.screenshot({ path: 'verification/screenshots/5_submission_modal.png' });
  } else {
      console.log('Button not visible');
  }

  await browser.close();
})();
